File: API/Helpers/UserParams.cs
namespace API.Helpers;

public class UserParams : PaginationParams
{
    public string? Gender { get; set; }
    public string? CurrentUsername { get; set; }
    public int MinAge { get; set; } = 0;
    public int MaxAge { get; set; } = 5;
    public string OrderBy { get; set; } = "lastActive";
    
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge = 0;
    maxAge = 5;
    pageNumber = 1;
    pageSize = 20;
    orderBy = 'lastActive';

    constructor(user: User) {
        if (user.roles.includes('Admin'))
            this.gender = '';
        else    
            this.gender = user.gender === 'female' ? 'male' : 'female'
    }
}